File: binary-search/binary_search.py
def search(self, nums: list, target: int) -> int:
    low = 0
    high = len(nums) - 1

    while low <= high:
        mid = (low + high) // 2
        guess = nums[mid]

        if guess == target:
            return mid
        elif guess < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1

File: binary-search/binary_search.js
var search = function(nums, target) {
    let low = 0;
    let high = nums.length - 1;
    let mid;
    let guess;

    while (low <= high){
        mid = Math.round((low + high) / 2);
        guess = nums[mid];

        if (guess === target) {
            return mid;
        } else if (guess < target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return -1;
};